### DISCUSSION TITLE
Change user displayName

### DISCUSSION THREAD

**Discussion by Godefroy** [nhost/nhost#1017] [Tips and Tricks]
I don't find how to change the current user `displayName`.
There are `changeEmail` and `changePassword` functions, but no `changeDisplayName` or `updateUserData`.
Am I missing something?

**Comment by plmercereau** [comment]
Hello Godefroy,
There's a [section in the documentation that explains how to get user information from a GraphQL query](https://docs.nhost.io/platform/authentication/users#get-user-information-using-graphql):

```graphql
query {
  user(id: "<user-id>") {
    displayName
  }
}
```

Don't forget to set your Hasura permissions accordingly. In most cases, you filter the `select` permissions of the `user` role with `{ id: { _eq: "x-hasura-user-id" } }`.

In the SDK, we implemented `changeEmail` and `changePassword` as they are linked to specific workflows (transactional email, password check, etc), and this is done by Hasura-auth's REST API, not GraphQL. 
The raw information is available through GraphQL, but we did not implement dedicated methods in the SDK as permissions and the extent of GraphQL operations varies from one application to another.

If you feel the documentation is too thin to explain this, don't hesitate to come up with a PR :)

**Comment by Godefroy** [comment]
Hi @plmercereau, thanks for your answer!

I managed to change the user displayName with this mutation:

```gql
mutation changeDisplayName($userId: uuid!, $displayName: String!) {
  updateUser(pk_columns: { id: $userId }, _set: { displayName: $displayName }) {
    id
    displayName
  }
}
```

However, I could not insert the user ID directly with "X-Hasura-User-id".

Also, displayName doesn't get updated on the frontend, I have to refresh the page.

**Reply by elitan** [reply]
> I could not insert the user ID directly with "X-Hasura-User-id".

What exactly do you mean here? What did you try to do with `x-hasura-user-id`?

Generally, this is how I would do it:

1. Use the same GraphQL mutation that you used.
2. Populate the `$userId` variable with the value of `const { id } = nhost.auth.getUser()`.
3. Set permissions in Hasura usina a custom check so users can only update their own `displayName`:

```json
{
   "id":{
      "_eq":"X-Hasura-User-Id"
   }
}
```

> displayName doesn't get updated on the frontend, I have to refresh the page.

This is related to how your frontend works and not something that Nhost (backend) manages.
